### ISSUE TITLE
Update to v0.46

### ISSUE THREAD

**Issue by okwme** [strangelove-ventures/packet-forward-middleware#21]
module needs to be updated to v0.46 for Rho upgrade.
current work in https://github.com/cosmos/gaia/pull/1436 blocked by this.
IG hub team will try updating but might need support

**Comment by jtieri** [comment]
Tried bumping the dep for the SDK to `v0.46.0-beta2` but looks like ibc-go v3 imports some things that are no longer present in 0.46 line of the SDK.

Here are some logs from after bumping the dep in `go.mod` and attempting to `go mod tidy`
```
github.com/strangelove-ventures/packet-forward-middleware/v2/router imports
        github.com/cosmos/ibc-go/v3/modules/apps/transfer/types tested by
        github.com/cosmos/ibc-go/v3/modules/apps/transfer/types.test imports
        github.com/tendermint/tendermint/state: module github.com/tendermint/tendermint@latest found (v0.35.4), but does not contain package github.com/tendermint/tendermint/state
github.com/strangelove-ventures/packet-forward-middleware/v2/router imports
        github.com/cosmos/ibc-go/v3/modules/core/04-channel/types tested by
        github.com/cosmos/ibc-go/v3/modules/core/04-channel/types.test imports
        github.com/cosmos/ibc-go/v3/testing/simapp imports
        github.com/cosmos/cosmos-sdk/x/auth/ante: package github.com/cosmos/cosmos-sdk/x/auth/ante provided by github.com/cosmos/cosmos-sdk at latest version v0.45.4 but not at required version v0.46.0-beta2
github.com/strangelove-ventures/packet-forward-middleware/v2/router imports
        github.com/cosmos/ibc-go/v3/modules/core/04-channel/types tested by
        github.com/cosmos/ibc-go/v3/modules/core/04-channel/types.test imports
        github.com/cosmos/ibc-go/v3/testing/simapp imports
        github.com/cosmos/cosmos-sdk/x/auth/client/rest: package github.com/cosmos/cosmos-sdk/x/auth/client/rest provided by github.com/cosmos/cosmos-sdk at latest version v0.45.4 but not at required version v0.46.0-beta2
github.com/strangelove-ventures/packet-forward-middleware/v2/router imports
        github.com/cosmos/ibc-go/v3/modules/core/04-channel/types tested by
        github.com/cosmos/ibc-go/v3/modules/core/04-channel/types.test imports
        github.com/cosmos/ibc-go/v3/testing/simapp imports
        github.com/cosmos/ibc-go/v3/modules/core/02-client/client imports
        github.com/cosmos/cosmos-sdk/types/rest: package github.com/cosmos/cosmos-sdk/types/rest provided by github.com/cosmos/cosmos-sdk at latest version v0.45.4 but not at required version v0.46.0-beta2
github.com/strangelove-ventures/packet-forward-middleware/v2/router imports
        github.com/cosmos/ibc-go/v3/modules/core/04-channel/types tested by
        github.com/cosmos/ibc-go/v3/modules/core/04-channel/types.test imports
        github.com/cosmos/ibc-go/v3/testing/simapp imports
        github.com/cosmos/ibc-go/v3/modules/core/02-client/client imports
        github.com/cosmos/cosmos-sdk/x/gov/client/rest: package github.com/cosmos/cosmos-sdk/x/gov/client/rest provided by github.com/cosmos/cosmos-sdk at latest version v0.45.4 but not at required version v0.46.0-beta2
github.com/strangelove-ventures/packet-forward-middleware/v2/router imports
        github.com/cosmos/ibc-go/v3/modules/core/04-channel/types tested by
        github.com/cosmos/ibc-go/v3/modules/core/04-channel/types.test imports
        github.com/cosmos/ibc-go/v3/testing/simapp imports
        github.com/cosmos/ibc-go/v3/testing/simapp/params imports
        github.com/cosmos/cosmos-sdk/x/auth/legacy/legacytx: package github.com/cosmos/cosmos-sdk/x/auth/legacy/legacytx provided by github.com/cosmos/cosmos-sdk at latest version v0.45.4 but not at required version v0.46.0-beta2
```

**Comment by DavidNix** [comment]
Following up because of renewed interest in this problem. 

TLDR; I still get the same error. 

Cosmos SDK `v0.46.0-beta2` is still the latest release for `v0.46`. 

This `beta2` release removed the packages:
* github.com/cosmos/cosmos-sdk/x/gov/client/rest
* github.com/cosmos/cosmos-sdk/x/auth/client/rest
* github.com/cosmos/cosmos-sdk/x/auth/legacy/legacytx
* github.com/cosmos/cosmos-sdk/x/auth/ante
* github.com/cosmos/cosmos-sdk/types/rest

However, `ibc-go/v3` still depends on the above packages. And this repo, `packet-forward-middleware`, depends on `ibc-go/v3`. 

My recommendation is `ibc-go/v3` needs to update to remove dependencies the deleted cosmos packages. I checked tip of upstream, and unfortunately these packages are still in use by ibc-go.

We could fork `ibc-go/v3` but we risk destabilizing the ibc modules.

**Comment by DavidNix** [comment]
@boojamya fyi ☝️

**Comment by DavidNix** [comment]
Per Interchain, try these branches/forks.

* https://github.com/oopcode/ibc-go/tree/chore/46-beta2
* https://github.com/cosmos/ibc-go/tree/carlos/upgrade-sdk-0.46-tendermint-0.35

**Comment by jtieri** [comment]
Updated to SDK v0.46.0 in a branch that I will open a PR from. This will update the Tendermint version, the ibc-go version, as well as the Go version to 1.18.
